lib/model.js

```diff
@@ -3747,6 +3747,24 @@ class Model {
   }
 
   /**
+   * Helper method to determine if a instance is "soft deleted".  This is
+   * particularly useful if the implementer renamed the `deletedAt` attribute
+   * to something different.  This method requires `paranoid` to be enabled.
+   *
+   * @returns {Boolean}
+   */
+  isSoftDeleted() {
```

Comment:
@felixfbecker The logic is meant to get the value of the property being used for soft delete (which by default is `deletedAt`. 

@sushantdhiman I didn't consider future deletes. I will update the logic to return true if the value of the soft delete property is same as or before now.